### PR TITLE
Keep organized for color filter

### PR DESCRIPTION
**Pull request by iory**
Add keep organized to hsi_color_filter.
To solve #1751

### CODE REVIEW

**Comment by iory** [comment]
hsi_color_filter

**Comment by Horisu** [comment]
it seems you didn't add color_filter.h

**Comment by iory** [comment]
oh, missed

**Comment by Horisu** [comment]
color_filter_nodelet.cppの249行目からの
```
    srv_ = boost::make_shared <dynamic_reconfigure::Server<Config> > (*pnh_);
    typename dynamic_reconfigure::Server<Config>::CallbackType f =
      boost::bind (&ColorFilter::configCallback, this, _1, _2);
    srv_->setCallback (f);

    filter_instance_ = pcl::ConditionalRemoval<pcl::PointXYZRGB>(true);
    filter_instance_.setKeepOrganized(keep_organized_);
```
を逆にして
```
    filter_instance_ = pcl::ConditionalRemoval<pcl::PointXYZRGB>(true);
    filter_instance_.setKeepOrganized(keep_organized_);

    srv_ = boost::make_shared <dynamic_reconfigure::Server<Config> > (*pnh_);
    typename dynamic_reconfigure::Server<Config>::CallbackType f =
      boost::bind (&ColorFilter::configCallback, this, _1, _2);
    srv_->setCallback (f);
```
こうしないと最初の設定がfilter_instanceに読み込まれないようで、rqt_reconfigure辺りでconfigCallbackを呼んであげないと
[pcl::ConditionalRemoval::applyFilter] not capable!
と出続けます。

**Comment by iory** [comment]
おお，ありがとうございます！

File: jsk_pcl_ros/include/jsk_pcl_ros/color_filter.h [hunk 1]
```diff
@@ -82,6 +82,7 @@ namespace jsk_pcl_ros
     virtual void unsubscribe();
     boost::shared_ptr<message_filters::Synchronizer<SyncPolicy> >sync_;
 
+    bool keep_organized_;
```
Comment (wkentaro) [inline jsk_pcl_ros/include/jsk_pcl_ros/color_filter.h]: Does `keep_organized_` need to be the attribute of class?
It seems only used in `onInit` function.

Comment (iory) [inline jsk_pcl_ros/include/jsk_pcl_ros/color_filter.h]: hmm... It doesn't need.

File: jsk_pcl_ros/src/color_filter_nodelet.cpp [hunk 2]
```diff
@@ -241,16 +241,19 @@ namespace jsk_pcl_ros
   {
     ConnectionBasedNodelet::onInit();
 
-    filter_instance_ = pcl::ConditionalRemoval<pcl::PointXYZRGB>(true);
     updateCondition();
+    bool keep_organized_;
```
Comment (wkentaro) [inline jsk_pcl_ros/src/color_filter_nodelet.cpp]: Please remove `_` suffix. 
Probably it is the naming rule for class attribute. https://google.github.io/styleguide/cppguide.html#Variable_Names

Comment (iory) [inline jsk_pcl_ros/src/color_filter_nodelet.cpp]: Fixed

**Comment by wkentaro** [comment]
lgtm

**Comment by iory** [comment]
thx!
2016/06/18 12:03 "Kentaro Wada" <notifications@github.com>:

**Comment by wkentaro** [comment]
But if you have time, please add gui program in the sample like

-  https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_perception/sample/sample_multiply_mask_image.launch#L15-L28
- https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_perception/test/multiply_mask_image.test#L3-L5

**Comment by iory** [comment]
Added it

**Comment by iory** [comment]
Please merge this.

**Pull request merged**